### PR TITLE
[FEATURE]: Design custom toast component 

### PR DESCRIPTION
**Pull request by HanikJain**
## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Styling change
- [ ] Documentation update
- [ ] Refactoring
- [ ] Chore
  
## Checklist:
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release`, `style` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Added unit tests for the changes made (if required)
- [x] Have you made sure the commit messages meets the guidelines?
- [x] Added relevant screenshots for the changes
- [x] Have you tested the changes on local/staging?
- [x] Have you made sure the code you have written follows the best practises to the best of your knowledge?

## PR Description: Implementing Custom Toast Component with ChakraUI

Overview:

This PR introduces a custom toast component along with a custom hook utilizing ChakraUI to enhance our application's user feedback experience. 

Changes:
1. New Component: Toast
   - Added a new custom toast component `Toast`.
   - Implemented functionalities such as customizable toast messages, duration, and positioning for enhanced user interaction.
   - Implemented different status `success` `error`, `warning`, `info`

2. Custom Hook: useCustomToast
   - Developed a custom hook `useCustomToast` for using the `Toast` component throughout the application.
   - The hook provides a simple API for triggering toast notifications with customizable content and options.

## Screenshots:
Screenshot 2024-03-15 204657
Screenshot 2024-03-15 204737

### CODE REVIEW

**Comment by HanikJain** [comment]
Hey, I would like more visibility on this. Should it be a custom hook, like replacing the current useToast with useCustomToast and passing custom config to render the custom component? Alternatively, should I design a custom toast component that will be added to the current toast function, for example: toast({ render: () => (<CustomToast />) })?

@sumitd94 @macintushar

**Comment by HanikJain** [comment]
@sumitd94 I have added a custom hook implementation. Currently, I have only implemented it in SignIn.tsx. Please review the code structure and let me know if any changes are required

File: .env.example
Comment (sumitd94) [inline .env.example:1]: why is this deleted?

File: src/components/Toast/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    if (status === 'success') {
+      return <Icon as={CheckCircleIcon} color={color} marginRight="16px"  boxSize="20px" />;
+    } else {
+      return <Icon as={InfoIcon} color={color} marginRight="16px" boxSize="20px" />;
+    }
+    
+};
+
+const CustomToast: React.FC<ToastProps> = ({
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:31]: We can just rename this to `Toast`, no need of custom

File: src/components/Toast/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+const CustomToastIcon: React.FC<CustomToastIconProps> = ({status}) => {
+    const color: string = status ? `${status}.400` : 'info.400';
+
+    if (status === 'success') {
+      return <Icon as={CheckCircleIcon} color={color} marginRight="16px"  boxSize="20px" />;
+    } else {
+      return <Icon as={InfoIcon} color={color} marginRight="16px" boxSize="20px" />;
+    }
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:27]:
```suggestion
    return <Icon as={status === 'success' ? CheckedCircleIcon : InfoIcon} color={color} marginRight="16px" boxSize="20px" />;
```

Comment (sumitd94) [reply src/components/Toast/index.tsx:27]: Also, where are we handling cases for `error`, `warning`?

File: src/components/Toast/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                borderColor={borderColor} 
+                width="400px"
+                paddingX="16px" 
+                paddingY="25px" 
+                borderRadius="8px" 
+                display="flex"
+                justifyContent="space-between"
+                >
+                    <Box display="flex"  justifyContent="flex-start" alignItems="center">
+                        < CustomToastIcon status={status} />
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:55]:
```suggestion
                        <CustomToastIcon status={status} />
```

File: src/components/Toast/index.tsx
Comment (sumitd94) [inline src/components/Toast/index.tsx:1]: I can still see double quotes and single quotes. Please run the lint and prettier to format the code

File: src/components/Toast/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                        <Box>
+                            {text}
+                        </Box>
+                    </Box> 
+                 
+                    <Button variant="unstyled" width="auto" onClick={onClose} marginX="16px"   >
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:61]:
```suggestion
                    <Button variant="unstyled" width="auto" onClick={onClose} marginX="16px">
```

File: src/components/Toast/index.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                        <Icon as={CloseIcon} color="black.100" boxSize="20px" />
+                    </Button>
+                    
+                     
+                    
+            </Box>
+        );  
+    }
+
+    <Box style={style}> < CustomToastIcon status={status} /> {text}</Box>
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:71]:
```suggestion
    <Box style={style}> < CustomToastIcon status={status} />{text}</Box>
```

File: .env.example
Comment (HanikJain) [reply .env.example:1]: I believe it has been renamed because of this cmd mv `.env.example .env`

File: src/components/Toast/index.tsx (hunk 1 above)
Comment (HanikJain) [reply src/components/Toast/index.tsx:31]: Only the component name or everything where customToast is used should be replaced with toast.

File: src/components/Toast/index.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Box, Button } from '@chakra-ui/react';
+import { CheckCircleIcon, InfoIcon, Icon, CloseIcon } from '@chakra-ui/icons';
+
+export type customToastStatus = 'success' | 'info' | 'warning' | 'error' | undefined;
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:5]: Can we have this as an enum?

Comment (sumitd94) [reply src/components/Toast/index.tsx:5]: `undefined` is not a valid one, if the status is not passed, treat it as `default`

File: src/components/Toast/index.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+interface CustomToastIconProps {
+  status?: customToastStatus;
+}
+interface ToastProps {
+  text?: string;
+  status?: customToastStatus;
+  style?: Record<string, string>;
+  onClose?: () => void;
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:14]: why are all these optional?

File: src/components/Toast/index.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      color={color}
+      marginRight='16px'
+      boxSize='20px'
+    />
+  );
+};
+
+const Toast: React.FC<ToastProps> = ({
+  text,
+  status = undefined,
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:31]:
```suggestion
  status = default,
```

File: src/components/Toast/index.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  style = {},
+  onClose = () => {},
+}) => {
+  if (status) {
+    const bg: string = `${status}.100`;
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:36]:
```suggestion
    const backgroundColor: string = `${status}.100`;
```

File: src/components/Toast/index.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        </Box>
+
+        <Button variant='unstyled' width='auto' onClick={onClose} marginX='16px'>
+          <Icon as={CloseIcon} color='black.100' boxSize='20px' />
+        </Button>
+      </Box>
+    );
+  }
+
+  <Box style={style}>
```
Comment (sumitd94) [inline src/components/Toast/index.tsx:62]: `style` variable naming is too vague here, can we be more specific please? Like `ToastContainerStyle`?

File: src/components/Toast/index.tsx (hunk 6 above)
Comment (HanikJain) [reply src/components/Toast/index.tsx:5]: What type of toast do I need to show as the default `success` `error`, `warning`, `info` @sumitd94

Comment (sumitd94) [reply src/components/Toast/index.tsx]: Info can be treated as the default one

**Review by sumitd94 — APPROVED** [review]
Looks good to me, thanks for the contribution :)
